src/antsibull/collection_meta.py

```diff
@@ -13,50 +13,85 @@
+def _convert_pypi_version(v: t.Any) -> t.Any:
+    if not isinstance(v, str):
+        raise ValueError(f"must be a string, got {v!r}")
+    if not v:
+        raise ValueError(f"must be a non-trivial string, got {v!r}")
+    version = PypiVer(v)
+    if len(version.release) != 3:
+        raise ValueError(
+            f"must be a version with three release numbers (e.g. 1.2.3, 2.3.4a1), got {v!r}"
+        )
+    return version
+
+
+PydanticPypiVersion = Annotated[PypiVer, BeforeValidator(_convert_pypi_version)]
+
+
+class RemovalInformation(p.BaseModel):
+    """
+    Stores metadata on when and why a collection will get removed.
+    """
+
+    model_config = p.ConfigDict(extra="ignore", arbitrary_types_allowed=True)
+
+    version: t.Union[int, t.Literal["TBD"]]
+    reason: t.Literal["deprecated", "considered-unmaintained", "renamed", "other"]
+    reason_text: str
+    announce_version: t.Optional[PydanticPypiVersion] = None
+    new_name: t.Optional[str] = None
+    discussion: t.Optional[p.HttpUrl] = None
+    redirect_replacement_version: t.Optional[int] = None
+
+
+class CollectionMetadata(p.BaseModel):
     """
     Stores metadata about one collection.
     """
 
-    changelog_url: str | None
-    collection_directory: str | None
-    repository: str | None
-    tag_version_regex: str | None
+    model_config = p.ConfigDict(extra="ignore")
 
-    def __init__(self, source: Mapping[str, t.Any] | None = None):
-        if source is None:
-            source = {}
-        self.changelog_url = source.get("changelog-url")
-        self.collection_directory = source.get("collection-directory")
-        self.repository = source.get("repository")
-        self.tag_version_regex = source.get("tag_version_regex")
+    changelog_url: t.Optional[str] = p.Field(alias="changelog-url", default=None)
+    collection_directory: t.Optional[str] = p.Field(
+        alias="collection-directory", default=None
+    )
+    repository: t.Optional[str] = None
+    tag_version_regex: t.Optional[str] = None
+    maintainers: list[str] = []
+    removal: t.Optional[RemovalInformation] = None
 
 
-class CollectionsMetadata:
+class CollectionsMetadata(p.BaseModel):
     """
     Stores metadata about a set of collections.
     """
 
-    data: dict[str, CollectionMetadata]
+    model_config = p.ConfigDict(extra="ignore")
+
+    collections: dict[str, CollectionMetadata]
 
-    def __init__(self, deps_dir: str | None):
-        self.data = {}
-        if deps_dir is not None:
-            collection_meta_path = os.path.join(deps_dir, "collection-meta.yaml")
-            if os.path.exists(collection_meta_path):
-                data = load_yaml_file(collection_meta_path)
-                if data and "collections" in data:
-                    for collection_name, collection_data in data["collections"].items():
-                        self.data[collection_name] = CollectionMetadata(collection_data)
+    @staticmethod
+    def load_from(deps_dir: str | None):
```

Comment:
```suggestion
    def load_from(deps_dir: str | os.PathLike[str] | None):
```

Reply:
We can also use `StrPath` here :)

Reply:
ed725859c9cc9035d594a1cb9452f1dafae79af1